airflow-core/src/airflow/jobs/triggerer_job_runner.py

```diff
@@ -604,74 +607,96 @@ def update_triggers(self, requested_trigger_ids: set[int]):
         adds them to the dequeues so the subprocess can actually mutate the running
         trigger set.
         """
-        render_log_fname = log_filename_template_renderer()
+        from airflow.models.dagbag import DBDagBag
 
-        known_trigger_ids = (
-            self.running_triggers.union(x[0] for x in self.events)
-            .union(self.cancelling_triggers)
-            .union(trigger[0] for trigger in self.failed_triggers)
-            .union(trigger.id for trigger in self.creating_triggers)
-        )
-        # Work out the two difference sets
-        new_trigger_ids = requested_trigger_ids - known_trigger_ids
-        cancel_trigger_ids = self.running_triggers - requested_trigger_ids
-        # Bulk-fetch new trigger records
-        new_triggers = Trigger.bulk_fetch(new_trigger_ids)
-        trigger_ids_with_non_task_associations = Trigger.fetch_trigger_ids_with_non_task_associations()
-        to_create: list[workloads.RunTrigger] = []
-        # Add in new triggers
-        for new_id in new_trigger_ids:
-            # Check it didn't vanish in the meantime
-            if new_id not in new_triggers:
-                log.warning("Trigger disappeared before we could start it", id=new_id)
-                continue
-
-            new_trigger_orm = new_triggers[new_id]
-
-            # If the trigger is not associated to a task, an asset, or a callback, this means the TaskInstance
-            # row was updated by either Trigger.submit_event or Trigger.submit_failure
-            # and can happen when a single trigger Job is being run on multiple TriggerRunners
-            # in a High-Availability setup.
-            if new_trigger_orm.task_instance is None and new_id not in trigger_ids_with_non_task_associations:
-                log.info(
-                    (
-                        "TaskInstance Trigger is None. It was likely updated by another trigger job. "
-                        "Skipping trigger instantiation."
-                    ),
-                    id=new_id,
-                )
-                continue
+        dag_bag = DBDagBag()
+        render_log_fname = log_filename_template_renderer()
 
-            workload = workloads.RunTrigger(
-                classpath=new_trigger_orm.classpath,
-                id=new_id,
-                encrypted_kwargs=new_trigger_orm.encrypted_kwargs,
-                ti=None,
-            )
-            if new_trigger_orm.task_instance:
-                log_path = render_log_fname(ti=new_trigger_orm.task_instance)
+        @provide_session
+        def create_workload(trigger: Trigger, session: Session = NEW_SESSION) -> workloads.RunTrigger | None:
+            if trigger.task_instance:
```

Comment:
Would early return for non TI related Trigger workload be better?
```suggestion
            if trigger.task_instance is None:
                return workloads.RunTrigger(
                    id=new_id,
                    classpath=trigger.classpath,
                    encrypted_kwargs=trigger.encrypted_kwargs,
                )
```